crates/sui-core/src/epoch/reconfiguration.rs

```diff
@@ -94,18 +105,69 @@ where
             .collect();
         let new_committee = Committee::new(next_epoch, votes);
         self.state.insert_new_epoch_info(&new_committee)?;
-        let new_net = Arc::new(AuthorityAggregator::new(
-            new_committee,
-            self.net.load().clone_inner_clients(),
-            self.gateway_metrics.clone(),
-        ));
-        self.net.store(new_net.clone());
-        // TODO: Also reconnect network if changed.
-        // This is blocked for now since we are not storing network info on-chain yet.
 
+        // Reconnect the network if we have an type of AuthorityClient that has a network.
+        if A::needs_network_recreation() {
+            let mut new_clients = BTreeMap::new();
+            let sui_system_state = self
+                .state
+                .get_sui_system_state_object()
+                .await
+                .map_err(|e| SuiError::GenericAuthorityError {
+                    error: e.to_string(),
+                })
+                .unwrap();
+            let next_epoch_validators = sui_system_state.validators.next_epoch_validators;
+
+            let mut net_config = mysten_network::config::Config::new();
+            net_config.connect_timeout = Some(Duration::from_secs(5));
+            net_config.request_timeout = Some(Duration::from_secs(5));
+            net_config.http2_keepalive_interval = Some(Duration::from_secs(5));
+
+            for validator in next_epoch_validators {
+                let net_addr: &[u8] = &validator.net_address.clone();
+                let str_addr =
+                    std::str::from_utf8(net_addr).map_err(|e| SuiError::GenericAuthorityError {
+                        error: e.to_string(),
+                    });
+                let address: Multiaddr = str_addr
+                    .unwrap()
+                    .parse()
+                    .map_err(|e: multiaddr::Error| SuiError::GenericAuthorityError {
+                        error: e.to_string(),
+                    })
+                    .unwrap();
+
+                let channel = net_config
+                    .connect_lazy(&address)
+                    .map_err(|e| SuiError::GenericAuthorityError {
+                        error: e.to_string(),
+                    })
+                    .unwrap();
+                let client: A = A::recreate(channel);
+                let name: &[u8] = &validator.name;
```

Comment:
These code seems similar to the one in sui-node, might be reusable if we make them a utility function somewhere

Reply:
It would be reusable if the `Validator` (used for this epoch) and `ValidatorMetadata` (used for next epoch) move objects were consolidated.

Reply:
The part of information you care about are all from metadata though.